homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -81,8 +78,8 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
 
     hap = HomematicipHAP(hass, entry)
-    hass.data[DOMAIN][entry.unique_id] = hap
 
+    entry.runtime_data = hap
```

Comment:
I think the runtime_data should be unset if async_setup fails, but this is an existing issue and should be addressed in follow-up PR